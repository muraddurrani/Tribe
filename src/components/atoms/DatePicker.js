import React, { useState } from 'react'
import { View, StyleSheet, TouchableHighlight } from 'react-native'
import { Input, Icon } from 'react-native-elements'
import TertiaryButton from './TertiaryButton'
import theme from '../../styles/theme'
import DateTimePicker from '@react-native-community/datetimepicker'

function DatePicker(props) {

  const [expand, setExpand] = useState(false)
  const [value, setValue] = useState('')

  return (
    <View style = {styles.container}>
      <TouchableHighlight
        underlayColor = {theme.colours.gray2}
        onPress = {() => setExpand(!expand)}
        >
        <Input
          containerStyle = {styles.input}
          editable = {false}
          value = {props.value ? props.value.toDateString().substring(4) : value}
          label = {props.label}
          placeholder = {props.placeholder}
          rightIcon = {<Icon name = 'calendar'/>}
        />
      </TouchableHighlight>
      {
        expand && (
          <DateTimePicker
            mode = "date"
            value = {new Date()}
            onChange = {(event, value) => {
              if (value !== undefined) {
                setExpand(false)
                setValue(value.toDateString().substring(4))
                props.onSelect(value)
              }
            }}
          />
        )
      }
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    width: 160
  },
  input: {
    width: 160
  }
})

export default DatePicker
