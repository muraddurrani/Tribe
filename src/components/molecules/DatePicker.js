import React, { useState } from 'react'
import { View, StyleSheet, TouchableHighlight } from 'react-native'
import { Input, Icon } from 'react-native-elements'
import DateTimePicker from '@react-native-community/datetimepicker'

import colours from '../../styles/colours'

function DatePicker({width, label, placeholder, onSelect}) {

  const [expand, setExpand] = useState(false)
  const [value, setValue] = useState('')

  return (
    <View style = {{width}}>
      <TouchableHighlight
        style = {styles.touchable}
        underlayColor = {colours.gray2}
        onPress = {() => setExpand(!expand)}
      >
        <Input
          containerStyle = {{width: width}}
          inputContainerStyle = {{width: width - 20}}
          editable = {false}
          value = {value}
          label = {label}
          placeholder = {placeholder}
          rightIcon = {<Icon name = 'calendar'/>}
        />
      </TouchableHighlight>
      {expand && (
        <DateTimePicker
          mode = "date"
          value = {new Date()}
          onChange = {(event, value) => {
            if (value !== undefined) {
              setExpand(false)
              setValue(value.toDateString().substring(4))
              onSelect(value)
            }
          }}
        />
      )}
    </View>
  )
}

const styles = StyleSheet.create({
  touchable: {
    borderRadius: 20
  }
})

export default DatePicker
