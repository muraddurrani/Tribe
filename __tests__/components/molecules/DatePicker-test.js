import React from 'react';
import renderer from 'react-test-renderer';
import DatePicker from '../../../src/components/molecules/DatePicker'

jest.useFakeTimers()

it('DatePicker renders correctly', () => {
  const tree = renderer
    .create(<DatePicker width = {160} label = "Date of Birth" placeholder = "Select D.O.B" onSelect = {console.log('DatePicker selected')}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});