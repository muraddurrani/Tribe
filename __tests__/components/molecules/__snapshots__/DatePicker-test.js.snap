// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker renders correctly 1`] = `
<View
  style={
    Object {
      "width": 160,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderRadius": 10,
      }
    }
  >
    <View
      style={
        Object {
          "height": 60,
          "paddingHorizontal": 10,
          "width": 160,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#86939e",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Date of Birth
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
            "width": 140,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          editable={false}
          placeholder="Select D.O.B"
          placeholderTextColor="#86939e"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "#242424",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginVertical": 4,
              "paddingRight": 4,
            }
          }
        >
          <View
            style={
              Object {
                "overflow": "hidden",
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#242424",
                        "fontSize": 24,
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="iconIcon"
                >
                  ?
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#ff190c",
            "fontSize": 12,
            "margin": 5,
          }
        }
      />
    </View>
  </View>
</View>
`;
